evalml/model_understanding/prediction_explanations/_user_interface.py

```diff
@@ -146,63 +194,86 @@ def make_dataframe(self, shap_values, normalized_values, pipeline_features, top_
+        super().__init__(top_k, include_shap_values, provenance)
         self.class_names = class_names
 
-    def make_text(self, shap_values, normalized_values, pipeline_features, top_k, include_shap_values=False):
+    def make_text(self, aggregated_shap_values, aggregated_normalized_values,
+                  shap_values, normalized_values, pipeline_features, original_features):
         strings = []
-        for class_name, class_values, normalized_class_values in zip(self.class_names, shap_values, normalized_values):
+        for class_name, class_values, normalized_class_values in zip(self.class_names, aggregated_shap_values,
+                                                                     aggregated_normalized_values):
             strings.append(f"Class: {class_name}\n")
-            table = _make_text_table(class_values, normalized_class_values, pipeline_features, top_k, include_shap_values)
+            table = _make_text_table(class_values, normalized_class_values, pipeline_features, original_features,
+                                     self.top_k, self.include_shap_values)
             strings += table.splitlines()
             strings.append("\n")
         return "\n".join(strings)
 
-    def make_dict(self, shap_values, normalized_values, pipeline_features, top_k, include_shap_values=False):
+    def make_dict(self, aggregated_shap_values, aggregated_normalized_values,
+                  shap_values, normalized_values, pipeline_features, original_features):
         json_output = []
-        for class_name, class_values, normalized_class_values in zip(self.class_names, shap_values, normalized_values):
-            rows = _make_rows(class_values, normalized_class_values, pipeline_features, top_k, include_shap_values,
-                              convert_numeric_to_string=False)
+        for class_index, class_name in enumerate(self.class_names):
+            rows = _make_rows(aggregated_shap_values[class_index], aggregated_normalized_values[class_index],
+                              pipeline_features, original_features, self.top_k,
+                              self.include_shap_values, convert_numeric_to_string=False)
             json_output_for_class = _rows_to_dict(rows)
+            drill_down = self.make_drill_down_dict(self.provenance, shap_values[class_index],
+                                                   normalized_values[class_index], pipeline_features, original_features,
+                                                   self.include_shap_values)
+            json_output_for_class["drill_down"] = drill_down
             json_output_for_class["class_name"] = _make_json_serializable(class_name)
             json_output.append(json_output_for_class)
         return {"explanations": json_output}
 
 
-def _make_single_prediction_shap_table(pipeline, pipeline_features, index_to_explain, top_k=3,
+def _make_single_prediction_shap_table(pipeline, pipeline_features, input_features, index_to_explain, top_k=3,
```

Comment:
Should we update this docstring to include `input_features` and `index_to_explain`?

Reply:
Yes!